From 5f2f7f7d52dc88cb0441cb738344994485fd5fdd Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 2 Nov 2017 11:16:35 -0700
Subject: [PATCH 1/2] Use char16_t instead of unsigned short for JSChar

Fixes

JSStringRef.cpp:75:30: error: invalid conversion from 'UChar* {aka char16_t*}' to 'const JSChar* {aka const short unsigned int*}' [-fpermissive]
|      return string->characters();
|             ~~~~~~~~~~~~~~~~~~^~

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 .../3rdparty/JavaScriptCore/Source/JavaScriptCore/API/JSStringRef.h    | 3 ++-
 netflix/src/platform/gibbon/scriptengine/jsc/JSC.h                     | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/netflix/3rdparty/JavaScriptCore/Source/JavaScriptCore/API/JSStringRef.h b/netflix/3rdparty/JavaScriptCore/Source/JavaScriptCore/API/JSStringRef.h
index 75d73c9..d05fe7d 100644
--- a/netflix/3rdparty/JavaScriptCore/Source/JavaScriptCore/API/JSStringRef.h
+++ b/netflix/3rdparty/JavaScriptCore/Source/JavaScriptCore/API/JSStringRef.h
@@ -32,6 +32,7 @@
 #include <stdbool.h>
 #endif
 #include <stddef.h> /* for size_t */
+#include <uchar.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -43,7 +44,7 @@ extern "C" {
 @typedef JSChar
 @abstract A Unicode character.
 */
-    typedef unsigned short JSChar;
+    typedef char16_t JSChar;
 #else
     typedef wchar_t JSChar;
 #endif
diff --git a/netflix/src/platform/gibbon/scriptengine/jsc/JSC.h b/netflix/src/platform/gibbon/scriptengine/jsc/JSC.h
index 5ff73bc..8109fc1 100644
--- a/netflix/src/platform/gibbon/scriptengine/jsc/JSC.h
+++ b/netflix/src/platform/gibbon/scriptengine/jsc/JSC.h
@@ -17,6 +17,7 @@
 #include "GibbonApplication.h"
 
 #include <API/JSValueRef.h>
+#include <uchar.h>
 
 typedef const struct OpaqueJSValue *JSValueRef;
 typedef struct OpaqueJSValue *JSObjectRef;
@@ -28,7 +29,7 @@ typedef struct OpaqueJSString *JSStringRef;
 // copied from JSStringRef.h
 #if !defined(WIN32) && !defined(_WIN32)                                 \
     && !((defined(__CC_ARM) || defined(__ARMCC__)) && !defined(__linux__)) /* RVCT */
-typedef unsigned short JSChar;
+typedef char16_t JSChar;
 #else
 typedef wchar_t JSChar;
 #endif
-- 
2.15.0

